tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/hive/HiveThriftClient.java

```diff
@@ -0,0 +1,235 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.tempto.internal.fulfillment.table.hive;
+
+import com.teradata.tempto.fulfillment.table.hive.statistics.ColumnStatistics;
+import com.teradata.tempto.fulfillment.table.hive.statistics.TableStatistics;
+import com.teradata.tempto.internal.fulfillment.table.TableName;
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsData;
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsDesc;
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsObj;
+import org.apache.hadoop.hive.metastore.api.Date;
+import org.apache.hadoop.hive.metastore.api.DateColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.Decimal;
+import org.apache.hadoop.hive.metastore.api.DecimalColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.DoubleColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.LongColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.StringColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore;
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.transport.TSocket;
+import org.apache.thrift.transport.TTransport;
+import org.apache.thrift.transport.TTransportException;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.primitives.Shorts.checkedCast;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class HiveThriftClient
+        implements AutoCloseable
+{
+    private final TTransport transport;
+    private final ThriftHiveMetastore.Client client;
+
+    public HiveThriftClient(String thriftHost, int thriftPort)
+    {
+        transport = new TSocket(thriftHost, thriftPort);
+        try {
+            transport.open();
+        }
+        catch (TTransportException e) {
+            throw new RuntimeException(e);
+        }
+        TBinaryProtocol protocol = new TBinaryProtocol(transport);
+        client = new ThriftHiveMetastore.Client(protocol);
+    }
+
+    void setStatistics(TableName tableName, TableStatistics tableStatistics)
+    {
+        try {
+            Table table = client.get_table(getSchema(tableName), tableName.getSchemalessNameInDatabase());
+            setRowsCount(tableName, tableStatistics, table);
+            try {
+                setColumnStatistics(tableName, tableStatistics, table, fieldSchema -> true);
+            } catch (TException ignore) {
+                transport.close();
```

Comment:
could you check exception message to verify if error we are seeing is actually about `date` type not being supported?